sihl_users/lib/service.ml

```diff
@@ -209,20 +209,25 @@ module User = struct
     in
     if not @@ Model.Token.is_valid_email_configuration token then
       Sihl_core.Fail.raise_bad_request "invalid confirmation token provided"
-      (* TODO do this with transaction *)
     else
-      let* () =
-        Repository.Token.update (Model.Token.inactivate token)
-        |> Sihl_core.Db.query_db_exn request
-      in
-      let* user =
-        Repository.User.get ~id:token.user |> Sihl_core.Db.query_db request
+      let* result =
+        Sihl_core.Db.query_db_with_trx request (fun connection ->
+            let* () =
+              Repository.Token.update (Model.Token.inactivate token)
+              |> Sihl_core.Db.query_db_exn request
+            in
+            let* user =
+              Repository.User.get ~id:token.user
+              |> Sihl_core.Db.query_db request
+            in
+            let user =
+              user |> Sihl_core.Fail.with_bad_request "invalid token provided"
+            in
+            Repository.User.update (Model.User.confirm user) connection)
       in
-      let user =
-        user |> Sihl_core.Fail.with_bad_request "invalid token provided"
-      in
-      Repository.User.update (Model.User.confirm user)
-      |> Sihl_core.Db.query_db_exn request
+      result
+      |> Sihl_core.Fail.with_bad_request "invalid token provided"
```

Comment:
@jerben Why is this line necessary? This was added in this diff

Reply:
@aronerben The new `Sihl_core.Db.query_db_with_trx` doesn't have an `_exn` version, I added it. Error handling is an open topic atm and not done very consistently.

Reply:
Ok!